### KOBUMANZI TRISHIA       M24B23/011
### MUNANURA SHANE          M24B23/031
### OWEN MBABAZI JOSEPH     M24B23/012
### ATURINDA BEINEMBABAZI   M24B23/042

In [1]:
import pandas as pd

In [2]:
Loan_Data= pd.read_csv('Loan_Approval_Data (1).csv')
Loan_Data


Loan_ID  Gender Married Dependents     Education Self_Employed  \
0    LP001002    Male      No          0      Graduate            No   
1    LP001003    Male     Yes          1      Graduate            No   
2    LP001005    Male     Yes          0      Graduate           Yes   
3    LP001006    Male     Yes          0  Not Graduate            No   
4    LP001008    Male      No          0      Graduate            No   
..        ...     ...     ...        ...           ...           ...   
609  LP002978  Female      No          0      Graduate            No   
610  LP002979    Male     Yes         3+      Graduate            No   
611  LP002983    Male     Yes          1      Graduate            No   
612  LP002984    Male     Yes          2      Graduate            No   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         NaN             360.0   
1               4583             1508.0       128.0             360.0   
2               3000                0.0        66.0             360.0   
3               2583             2358.0       120.0             360.0   
4               6000                0.0       141.0             360.0   
..               ...                ...         ...               ...   
609             2900                0.0        71.0             360.0   
610             4106                0.0        40.0             180.0   
611             8072              240.0       253.0             360.0   
612             7583                0.0       187.0             360.0   
613             4583                0.0       133.0             360.0   

     Credit_History Property_Area Loan_Status  
0               1.0         Urban           Y  
1               1.0         Rural           N  
2               1.0         Urban           Y  
3               1.0         Urban           Y  
4               1.0         Urban           Y  
..              ...           ...         ...  
609             1.0         Rural           Y  
610             1.0         Rural           Y  
611             1.0         Urban           Y  
612             1.0         Urban           Y  
613             0.0     Semiurban           N  

[614 rows x 13 columns]

##### 1.Marginal probability of any continuous variable
             Marginal probability for Applicant Income

In [3]:
#Creating bins
bins = [0,2000,4000,6000,8000,10000,20000,30000,40000,50000]
labels = ['0-2000','2000-4000','4000-6000','6000-8000','8000-10000','10000-20000','20000-30000','30000-40000','40000-50000']

#Creating a new column for ApplicantIncome_Bin
Loan_Data['ApplicantIncome_bin'] = pd.cut(Loan_Data['ApplicantIncome'], bins = bins, labels=labels, right=False)

##Calculating the Marginal Probabilities
ApplicantIncome_probabilities = Loan_Data['ApplicantIncome_bin'].value_counts(normalize=True).sort_index()
ApplicantIncome_probabilities

ApplicantIncome_bin
0-2000         0.065466
2000-4000      0.476268
4000-6000      0.229133
6000-8000      0.096563
8000-10000     0.047463
10000-20000    0.070376
20000-30000    0.008183
30000-40000    0.006547
40000-50000    0.000000
Name: proportion, dtype: float64

##### 2.Marginal Probability for any categorical variable
             Marginal probability for Self_Employed

In [4]:
Self_Employed_probavilities = Loan_Data['Self_Employed'].value_counts()/len(Loan_Data)
print(f"Marginal probabilities for Self_Employed are : \n {Self_Employed_probavilities}")


Marginal probabilities for Self_Employed are : 
 Self_Employed
No     0.814332
Yes    0.133550
Name: count, dtype: float64


##### 3.Probability that a "female" applicant has her Loan_Status reflecting "Y"

In [5]:
#Filtering the data for only female applicants whose loan status is Y
filtered_data = (Loan_Data['Gender'] == 'Female') & (Loan_Data['Loan_Status'] == 'Y')

#Calculating number of outcomes from the filtered data
Female_Y_count = Loan_Data[filtered_data]['Gender'].count()
Female_Y_count

#Calculating the probability for females with a loan status reflecting Y
Female_Y_probability=Female_Y_count/len(Loan_Data)
Female_Y_probability

#Calculating the Y probability
Y_count=Loan_Data[Loan_Data['Loan_Status']=="Y"]['Loan_Status'].count()
Y_probability=Y_count/len(Loan_Data)

#Calculating the final conditional probability of females given a loan status reflecting Y
Final_probability=Female_Y_probability/Y_probability
Final_probability
print(f"The probability of females with a loan status reflecting Y is {Final_probability}")

The probability of females with a loan status reflecting Y is 0.17772511848341233


##### 4.Explain the central tendency for any two continuous variables.

In [6]:
Loan_Data.describe()

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       614.000000         614.000000  592.000000         600.00000   
mean       5403.459283        1621.245798  146.412162         342.00000   
std        6109.041673        2926.248369   85.587325          65.12041   
min         150.000000           0.000000    9.000000          12.00000   
25%        2877.500000           0.000000  100.000000         360.00000   
50%        3812.500000        1188.500000  128.000000         360.00000   
75%        5795.000000        2297.250000  168.000000         360.00000   
max       81000.000000       41667.000000  700.000000         480.00000   

       Credit_History  
count      564.000000  
mean         0.842199  
std          0.364878  
min          0.000000  
25%          1.000000  
50%          1.000000  
75%          1.000000  
max          1.000000

Applicant Income<br>
Mean=5403.459283<br>
Median=3812.500000<br>

Loan Amount<br>
Mean=146.412162<br>
Median=128.000000<br>

##### 5.Explain the relationship between the variables "Gender" and "Loan Status"

In [7]:
#Running the chisquare test to first find out the relationship

#Generating a cross-tabulation between 'Gender' and 'Loan Status'
CrosstabResult=pd.crosstab(index=Loan_Data['Gender'],columns=Loan_Data['Loan_Status'])
print(CrosstabResult)

from scipy.stats import chi2_contingency
ChiSqResult = chi2_contingency(CrosstabResult)
print('The P-Value is:', ChiSqResult[1])

Loan_Status    N    Y
Gender               
Female        37   75
Male         150  339
The P-Value is: 0.7086529816451106


Since the P-Value >0.05 then Gender and Loan_Status are not correlated,